clients/src/main/java/org/apache/kafka/common/requests/RequestHeader.java

```diff
@@ -89,11 +90,16 @@ public ResponseHeader toResponseHeader() {
     }
 
     public static RequestHeader parse(ByteBuffer buffer) {
-        short apiKey = buffer.getShort();
-        short apiVersion = buffer.getShort();
-        Schema schema = Protocol.requestHeaderSchema(apiKey, apiVersion);
-        buffer.rewind();
-        return new RequestHeader(schema.read(buffer));
+        try {
+            short apiKey = buffer.getShort();
+            short apiVersion = buffer.getShort();
+            Schema schema = Protocol.requestHeaderSchema(apiKey, apiVersion);
+            buffer.rewind();
+            return new RequestHeader(schema.read(buffer));
+        } catch (Throwable  ex) {
+            throw new InvalidRequestException("Error parsing request header. Our best guess of the apiKey is: " +
```

Comment:
Actually we probably just need to change the catch for `SchemaException` with `InvalidRequestException`, which seems reasonable.